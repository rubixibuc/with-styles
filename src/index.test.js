import React from 'react';
import withStyles from './index';
import { mount } from 'enzyme';

describe('with-styles', () => {
  it('should render component with styles when initialized with object', () => {
    const Component = withStyles({
      div: {
        backgroundColor: '#000'
      }
    })(({ styles }) => <div style={styles.div} />);

    const MountedComponent = mount(<Component />);

    expect(MountedComponent).toMatchSnapshot();
  });

  it('should render component with styles when initialized with object', () => {
    const Component = withStyles(({ backgroundColor }) => ({
      div: {
        backgroundColor
      }
    }))(({ styles }) => <div style={styles.div} />);

    const MountedComponent = mount(<Component backgroundColor="#fff" />);

    expect(MountedComponent).toMatchSnapshot();
  });
});
