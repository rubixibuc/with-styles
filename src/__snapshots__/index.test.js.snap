// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with-styles should render component with styles when initialized with object 1`] = `
<Component>
  <Component
    styles={
      Object {
        "div": Object {
          "backgroundColor": "#000",
        },
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "#000",
        }
      }
    />
  </Component>
</Component>
`;

exports[`with-styles should render component with styles when initialized with object 2`] = `
<Component
  backgroundColor="#fff"
>
  <Component
    backgroundColor="#fff"
    styles={
      Object {
        "div": Object {
          "backgroundColor": "#fff",
        },
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "#fff",
        }
      }
    />
  </Component>
</Component>
`;
